projects/fusion-ui/components/chip-filters/common/base/chip-filters.base.component.ts

```diff
@@ -152,22 +167,27 @@ export abstract class ChipFiltersBaseComponent implements AfterViewInit, OnDestr
     }
 
     private addChipFilter(option: DropdownOption): void {
-        this.chipFilters.toArray().forEach(chip => {
-            const isSelected = this.addedFilters.some(selectedChip => selectedChip.id === chip['id']);
-            if (chip['id'] === option.id && !isSelected && chip.mode === 'dynamic') {
-                chip['isVisible'] = true;
-                const newSelection = {
-                    id: option.id,
-                    value: option,
-                    isSelected: chip.selected
-                };
-                this.addedFilters = [...this.addedFilters, newSelection];
-                this.reduceSelectedFiltersOptions();
-                chip.apiBase.open();
-                this.cdr.markForCheck();
-            } else {
-                this.addFilterControl.reset();
-            }
+        this.onDynamicChipSelect.emit(option);
+
+        // need run in next tic because selected filter must be rendered in DOM
+        setTimeout(() => {
```

Comment:
it's not the best solution.
when new chip will insert the DOM 
this @ContentChildren(ChipFilterComponent) set chipFilters(value: QueryList<ChipFilterComponent>)
will be trigger. then you need to work with it and not run it under setTimeout

Reply:
Ok.